### PR TITLE
Fix ClassCastException

### LINKED ISSUE #1880: Boat erorr
### Issue Description

somthing damage some boat and crash chunks

### OS and Versions

* Nukkit Version: 1.0 dev 

* Java Version: 
```
8
```

* Host Configuration: 

| Item | Value |
|:----:|:-----:|
| Host OS | ?? |  
| Memory(RAM) | ?? GB | 
| Storage Size | ?? GB | 
| Storage Type | ?? | 
| CPU Type | ?? | 
| CPU Core Count | ?? cores ?? threads | 
| Upstream Bandwidth | ?? | 

* Client Configuration: 

| Item | Value |
|:----:|:-----:|
| Client Edition | ?? | 
| Client Version | ?? | 

### Crashdump, Backtrace or Other Files

```
<!--- Copy and paste your backtrace here... -->
```

### PR DESCRIPTION
**Pull request by xISRAPILx**
fix #1880

### CODE REVIEW

**Comment by NycuRO** [comment]
Tested?

**Comment by xISRAPILx** [comment]
Yes @NycuRO

**Comment by larryTheCoder** [comment]
By the way you can also do like this
```java
    @Override
    public boolean attack(EntityDamageEvent source) {
        if (invulnerable) {
            return false;
        } else {
            performHurtAnimation((int) source.getFinalDamage());

            Object damager = source instanceof EntityDamageByEntityEvent
                    ? ((EntityDamageByEntityEvent) source).getDamager()
                    : source instanceof EntityDamageByBlockEvent
                            ? ((EntityDamageByBlockEvent) source).getDamager()
                            : null;
            boolean instantKill = damager != null && damager instanceof Player && ((Player) damager).isCreative();
            if (instantKill || getDamage() > 40) {
                if (linkedEntity != null) {
                    mountEntity(linkedEntity);
                }

                if (instantKill && (!hasCustomName())) {
                    kill();
                } else {
                    if (level.getGameRules().getBoolean("doEntityDrops")) {
                        this.level.dropItem(this, new ItemBoat());
                    }
                    close();
                }
            }
        }

        return true;
    }
```
Reason why is when the lava is attacking the boat the boat wont get the damage

**Comment by xISRAPILx** [comment]
There is no difference in my opinion. Lava will still kill instantly, this is an extra load. @larryTheCoder

**Comment by larryTheCoder** [comment]
Okay, @PeratX you can merge this

**Comment by PeratX** [comment]
Something wrong with your coding format.

**Comment by xISRAPILx** [comment]
If do as you do - more checks are obtained = more load @PeratX

File: src/main/java/cn/nukkit/entity/item/EntityBoat.java [hunk 1]
```diff
@@ -94,23 +94,29 @@ public boolean attack(EntityDamageEvent source) {
             return false;
         } else {
             performHurtAnimation((int) source.getFinalDamage());
+            if(source instanceof EntityDamageByEntityEvent) {
+                Entity damager = ((EntityDamageByEntityEvent) source).getDamager();
+                boolean instantKill = damager instanceof Player && ((Player) damager).isCreative();
+                if (instantKill || getDamage() > 40) {
+                    if (linkedEntity != null) {
+                        mountEntity(linkedEntity);
```
Comment (NycuRO) [inline src/main/java/cn/nukkit/entity/item/EntityBoat.java]: this. ?

File: src/main/java/cn/nukkit/entity/item/EntityBoat.java [hunk 2]
```diff
@@ -94,23 +94,29 @@ public boolean attack(EntityDamageEvent source) {
             return false;
         } else {
             performHurtAnimation((int) source.getFinalDamage());
+            if(source instanceof EntityDamageByEntityEvent) {
```
Comment (NycuRO) [inline src/main/java/cn/nukkit/entity/item/EntityBoat.java]: reformat

File: src/main/java/cn/nukkit/entity/item/EntityBoat.java [hunk 3]
```diff
@@ -94,23 +94,29 @@ public boolean attack(EntityDamageEvent source) {
...
+                    } else {
+                        if (level.getGameRules().getBoolean("doEntityDrops")) {
+                            this.level.dropItem(this, new ItemBoat());
+                        }
+                        close();
                     }
-                    close();
                 }
-            }
+            }else{
```
Comment (NycuRO) [inline src/main/java/cn/nukkit/entity/item/EntityBoat.java]: reformat

**Comment by sero583** [comment]
@xISRALILx Do you know what formatting is?

**Comment by xISRAPILx** [comment]
yes =) @sero583 I'm not familiar with your code style.

**Comment by sero583** [comment]
The best style in my opinon is this one:
``if(statement==check) {

} else if(statement2==check2) {

} else {

}``

**Pull request closed without merging**